File: miami/csa174/exams/Niven.java

public final class Niven {

	public static boolean isNiven( int i ) {
		int digitSum = 0;
		
		int number = i;
		while( number >= 10 ) {
			digitSum += number % 10;
			number /= 10;
		}
		digitSum += number;
		
		return i % digitSum == 0;
	}

	public static void main( String[] args ) {
		int i = Integer.parseInt( args[0] );
		for( int j = 1; j <= i; j++ ) {
			System.out.printf("%5d -> %s%n", j, isNiven(j)+"" );
		}
	}
	

}

File: uc/computerScience2/spring2014/materials/exams/finalExamCode/niven.cpp
#include <iostream>
#include <cstdlib>
using namespace std;

int sumOfDigits(int number) {
  if (number < 10) {
    return number;
  }
  return (number % 10) + sumOfDigits(number / 10);
}

bool isNiven(int number) {
  if (number < 1) {
    return false;
  }
  int digitSum = sumOfDigits(number);
  return number % digitSum == 0; 
}

int main() {
  for (int i = 0; i <= 50; i++) {
    cout << i << " " << isNiven(i) << endl;
  }
}